### PR TITLE
tests: add typing

### PR DESCRIPTION
**Pull request by joelynch**
Been having some trouble reading tests without type annotations.

* add missing types to test methods
* remove unused fixtures `fixture_utils_http_request_list`
* just use `Mock` instead of `SimpleNamespace`, it is better for typing (mypy understands that `Mock` is a subtype / supertype of every type similar to `Any`).

### CODE REVIEW

File: tests/unit/coordinator/plugins/cassandra/test_plugin.py [hunk 1]
```diff
@@ -7,52 +7,53 @@
...
     mocker.patch.object(plugin, "run_subop")
 
     step = plugin.CassandraSubOpStep(op=ipc.CassandraSubOp.stop_cassandra)
-    cluster = None
-    context = None
-    result = await step.run_step(cluster, context)
+    cluster = Mock()
+    context = Mock()
+    result = await step.run_step(cluster, context)  # type: ignore[func-returns-value]
     assert result is None
```
Comment (kmichel-aiven) [inline tests/unit/coordinator/plugins/cassandra/test_plugin.py:33]: The test disables the static check for "returns None" and then assert that the return is None.
I think we can just remove the assert and the ignore.

File: tests/unit/coordinator/plugins/cassandra/test_plugin.py [hunk 2]
```diff
@@ -7,52 +7,53 @@
...
-async def test_step_cassandrasubop(mocker):
+async def test_step_cassandrasubop(mocker: MockerFixture) -> None:
     mocker.patch.object(plugin, "run_subop")
 
     step = plugin.CassandraSubOpStep(op=ipc.CassandraSubOp.stop_cassandra)
-    cluster = None
-    context = None
-    result = await step.run_step(cluster, context)
+    cluster = Mock()
+    context = Mock()
```
Comment (kmichel-aiven) [inline tests/unit/coordinator/plugins/cassandra/test_plugin.py:31]: cluster and context are super easy to create, no need to take risks and use mocks.

```python
cluster = Cluster(nodes=[])
context = StepsContext()
```

(same in other tests in the same file).

File: tests/unit/coordinator/plugins/cassandra/test_plugin.py [hunk 3]
```diff
@@ -7,52 +7,53 @@
...
+async def test_step_cassandra_validate_configuration(mocker: MockerFixture, success: bool) -> None:
     step = plugin.ValidateConfigurationStep(nodes=[])
-    context = None
+    context = Mock()
     if success:
-        cluster = SimpleNamespace(nodes=[])
-        result = await step.run_step(cluster, context)
+        cluster = Mock(nodes=[])
+        result = await step.run_step(cluster, context)  # type: ignore[func-returns-value]
         assert result is None
```
Comment (kmichel-aiven) [inline tests/unit/coordinator/plugins/cassandra/test_plugin.py:44]: Here too, we can just let mypy check the return type.

File: tests/unit/coordinator/plugins/cassandra/test_restore_steps.py [hunk 4]
```diff
@@ -93,9 +94,9 @@ def get_result(cl):
     step = restore_steps.StartCassandraStep(
         replace_backup_nodes=replace_backup_nodes, override_tokens=override_tokens, cassandra_nodes=[_configuration_node(1)]
     )
-    context = SimpleNamespace(get_result=get_result)
-    cluster = SimpleNamespace(nodes=nodes)
-    result = await step.run_step(cluster, context)
+    context = Mock(get_result=get_result)
+    cluster = Mock(nodes=nodes)
+    result = await step.run_step(cluster, context)  # type: ignore[func-returns-value]
```
Comment (kmichel-aiven) [inline tests/unit/coordinator/plugins/cassandra/test_restore_steps.py:99]: Same as the other test files, no need to use mocks and dynamically test static types.

File: tests/unit/coordinator/test_cleanup.py [hunk 5]
```diff
@@ -6,14 +6,26 @@
...
+def _run(
+    *,
+    client: TestClient,
+    populated_mstorage: MultiRohmuStorage,
+    app: FastAPI,
+    fail_at: int | None = None,
+    retention: ipc.Retention,
+    exp_jsons: int,
+    exp_digests: int,
+):
```
Comment (kmichel-aiven) [inline tests/unit/coordinator/test_cleanup.py:28]: `-> None`

File: tests/unit/coordinator/test_cleanup.py [hunk 6]
```diff
@@ -44,7 +56,7 @@ def _run(*, client, populated_mstorage, app, fail_at=None, retention, exp_jsons,
 
 
 @pytest.mark.parametrize("fail_at", FAILS)
-def test_api_cleanup_flow(fail_at, client, populated_mstorage, app):
+def test_api_cleanup_flow(fail_at: int | None, client: TestClient, populated_mstorage: MultiRohmuStorage, app: FastAPI):
```
Comment (kmichel-aiven) [inline tests/unit/coordinator/test_cleanup.py:59]: `-> None`

File: tests/unit/coordinator/test_cleanup.py [hunk 7]
```diff
@@ -69,7 +81,9 @@ def test_api_cleanup_flow(fail_at, client, populated_mstorage, app):
         (ipc.Retention(maximum_backups=1, keep_days=10000), 1, 1),
     ],
 )
-def test_api_cleanup_retention(data, client, populated_mstorage, app):
+def test_api_cleanup_retention(
+    data: tuple[ipc.Retention, int, int], client: TestClient, populated_mstorage: MultiRohmuStorage, app: FastAPI
+):
```
Comment (kmichel-aiven) [inline tests/unit/coordinator/test_cleanup.py:86]: `-> None`

File: tests/unit/coordinator/test_restore.py [hunk 8]
```diff
@@ -173,7 +176,9 @@ def match_clear(request: httpx.Request) -> Optional[httpx.Response]:
         (["foo", "foo", "bar", "bar"], ["1", "3", "3", "3"], None, pytest.raises(exceptions.InsufficientNodesException)),
     ],
 )
-def test_node_to_backup_index(node_azlist, backup_azlist, expected_index, exception):
+def test_node_to_backup_index(
+    node_azlist: list[str], backup_azlist: list[str], expected_index: list[int], exception: ContextManager
+):
```
Comment (kmichel-aiven) [inline tests/unit/coordinator/test_restore.py:181]: `-> None`

File: tests/unit/node/test_node_cassandra.py [hunk 9]
```diff
@@ -8,22 +8,27 @@
...
-def fixture_astacus_node_cassandra() -> None:
+def fixture_astacus_node_cassandra() -> ModuleType:
     return pytest.importorskip("astacus.node.cassandra")
 
 
 class CassandraTestEnv(CassandraTestConfig):
     cassandra_node_config: CassandraNodeConfig
 
-    def __init__(self, *, app, client, mocker, tmpdir):
+    def __init__(self, *, app, client, mocker, tmpdir: py.path.local):
```
Comment (kmichel-aiven) [inline tests/unit/node/test_node_cassandra.py:31]: `-> None`

File: tests/unit/node/test_node_cassandra.py [hunk 10]
```diff
@@ -53,14 +58,14 @@ def setup_cassandra_node_config(self):
...
-def fixture_ctenv(app, client, mocker, tmpdir):
+def fixture_ctenv(app: FastAPI, client: TestClient, mocker: MockerFixture, tmpdir: py.path.local) -> CassandraTestEnv:
     return CassandraTestEnv(app=app, client=client, mocker=mocker, tmpdir=tmpdir)
 
 
 @pytest.mark.parametrize(
     "subop", set(ipc.CassandraSubOp) - {ipc.CassandraSubOp.get_schema_hash, ipc.CassandraSubOp.restore_sstables}
 )
-def test_api_cassandra_subop(app, ctenv, mocker, subop):
+def test_api_cassandra_subop(app: FastAPI, ctenv: CassandraTestEnv, mocker: MockerFixture, subop: ipc.CassandraSubOp):
```
Comment (kmichel-aiven) [inline tests/unit/node/test_node_cassandra.py:68]: `-> None`

File: tests/unit/node/test_node_cassandra.py [hunk 11]
```diff
@@ -124,7 +129,9 @@ def test_api_cassandra_subop(app, ctenv, mocker, subop):
 
 
 @pytest.mark.parametrize("fail", [True])
-def test_api_cassandra_get_schema_hash(ctenv, fail, mocker, astacus_node_cassandra):
+def test_api_cassandra_get_schema_hash(
+    ctenv: CassandraTestEnv, fail: bool, mocker: MockerFixture, astacus_node_cassandra: ModuleType
+):
```
Comment (kmichel-aiven) [inline tests/unit/node/test_node_cassandra.py:134]: `-> None`

**Review by kmichel-aiven — CHANGES_REQUESTED** [review]

File: astacus/common/rohmustorage.py [hunk 12]
```diff
@@ -208,7 +208,7 @@ class MultiRohmuStorage(MultiStorage[RohmuStorage]):
     def __init__(self, *, config: RohmuConfig) -> None:
         self.config = config
 
-    def get_storage(self, name: str) -> RohmuStorage:
+    def get_storage(self, name: str | None) -> RohmuStorage:
```
[the remark this answers is not in the document]
Comment (joelynch) [reply astacus/common/rohmustorage.py:211]: Hmm are you sure about this?  I think the type widening makes sense in this case, it is matching the type of `RohmuStorage.__init__` which this method just forwards to.

File: tests/unit/common/test_op.py [hunk 13]
```diff
@@ -11,18 +11,19 @@
 from astacus.common.statsd import StatsClient
 from starlette.background import BackgroundTasks
 from starlette.datastructures import URL
+from unittest.mock import Mock
 
 import pytest
 
 
-class MockOp:
+class MockOp(op.Op):
```
[the remark this answers is not in the document]
Comment (joelynch) [reply tests/unit/common/test_op.py:19]: Good point, I can just use Op directly

File: tests/unit/common/test_utils.py [hunk 14]
```diff
@@ -173,7 +175,7 @@ def test_open_path_with_atomic_rename(tmpdir):
     with pytest.raises(AssertionError):
         with utils.open_path_with_atomic_rename(f3_path):
             assert False
-    assert not Path(f3_path).exists()
+    assert not Path(f3_path).exists()  # type: ignore[unreachable]
```
[the remark this answers is not in the document]
Comment (joelynch) [reply tests/unit/common/test_utils.py:178]: I'll change the exception type, but also this is a mypy bug in this version of mypy.  It does not work well with `pytest.raises`.  I've added a comment.

[the remark this answers is not in the document]
Comment (kmichel-aiven) [reply tests/unit/common/test_utils.py:183]: I'd change this to a plain try/except/else (with the not exist in the except branch and an assert False in the else branch), that wouldn't require mypy to know about pytest special things.

**Review by kmichel-aiven — APPROVED** [review]

**Pull request merged**